core/src/upgrade/apply.rs

```diff
@@ -204,38 +204,30 @@ where
     }
 }
 
-/// Wraps around a `UpgradeInfo` and satisfies the requirement of `listener_select_proto`.
-struct UpgradeIntoProtocolsIterWrap<U>(U);
+// Wraps around a `UpgradeInfo` and satisfies the requirement of `listener_select_proto`.
```

Comment:
Nit; something being private doesn't mean we should doc-comment it; the doc-comment won't show up in the docs but can still help people developers using the RLS and are hacking on libp2p
```suggestion
/// Wraps around a `UpgradeInfo` and satisfies the requirement of `listener_select_proto`.
```